src/ml_gym/util/util.py

```diff
@@ -191,15 +202,26 @@ def update_dataset_details(exp_config: dict) -> DatasetDetails:
                     obj (DatasetDetails): initialized dataset details object.
             """
             try:
-                dataset_splits = {"split_config": exp_config["dataset_iterators"]["config"]["split_configs"], 
-                              "splits_percentage": exp_config["splitted_dataset_iterators"]["config"]["split_configs"]}
+                splits_percentage = []
+                split_subscription = set()
+                find_key = "dataset_iterator"
+                for key in exp_config:
+                    if find_key in key and exp_config[key]["component_type_key"] == "SPLITTED_DATASET_ITERATORS":
+                        splits_percentage.append(exp_config[key]["config"]["split_configs"])
+                        for req in exp_config[key]["requirements"]:
+                            split_subscription.update(req["subscription"])
+                    elif exp_config[key]["component_type_key"] == "DATASET_ITERATORS":
+                        split_config = exp_config["dataset_iterators"]["config"]["split_configs"] if "split_configs" in exp_config["dataset_iterators"]["config"] else None
```

Comment:
still hardcoded paths / keys here

Reply:
Yeah, I forgot to remove it, the code has been changed to this now:
 split_config = exp_config[key]["config"]["split_configs"] if "split_configs" in exp_config[key]["config"] else None

The other hardcoding is internal structure, I assume that the internal structure keys are very specific and cannot be changed right? So technically no hardcoding, the path to be explored is static and is what is expected in the config.